server/src/main/java/org/elasticsearch/plugins/internal/DocumentSizeObserver.java

```diff
@@ -8,50 +8,39 @@
 
 package org.elasticsearch.plugins.internal;
 
+import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.xcontent.XContentParser;
 
 /**
  * An interface to allow wrapping an XContentParser and observe the events emitted while parsing
  * A default implementation returns a noop DocumentSizeObserver
  */
-public interface DocumentSizeObserver {
+public interface DocumentSizeObserver extends NormalizedDocumentSize {
     /**
      * a default noop implementation
      */
-    DocumentSizeObserver EMPTY_INSTANCE = new DocumentSizeObserver() {
+    DocumentSizeObserver NOOP = new DocumentSizeObserver() {
         @Override
-        public XContentParser wrapParser(XContentParser xContentParser) {
-            return xContentParser;
+        public long ingestedBytes() {
+            return 0;
         }
 
         @Override
-        public long normalisedBytesParsed() {
+        public long storedBytes() {
             return 0;
         }
-
     };
 
     /**
-     * Decorates a provided xContentParser with additional logic (gather some state).
-     *
-     * @param xContentParser to be decorated
-     * @return a decorator xContentParser
+     * Optionally decorates an xContentParser with additional logic.
      */
-    XContentParser wrapParser(XContentParser xContentParser);
+    default XContentParser wrapParser(XContentParser xContentParser) {
+        return xContentParser;
+    }
 
     /**
-     * Returns the state gathered during parsing
-     *
-     * @return a number representing a state parsed
+     * May enrich the index request with the number of bytes observed when parsing a document.
      */
-    long normalisedBytesParsed();
+    default void setNormalisedBytesParsedOn(IndexRequest indexRequest) {}
```

Comment:
This is very awkward; I understand we have the need to decide if we want to set the bytes, set 0, etc. based on the context, but it looks weird.

Maybe changing name and making this a function (return IndexRequest) would be better, but still an "Observer" does not look like the right place for this.

Reply:
I've removed this, I don't think it's necessary ... we can just always set the bytes on the IndexRequest, in some cases this just won't be read anymore.